text/0000-bundler-version-locking.md

```diff
@@ -0,0 +1,193 @@
+- Feature Name: bundler_version_locking
+- Start Date: 2020-09-01
+- RFC PR: (leave this empty)
+- Bundler Issue: (leave this empty)
+
+# Summary
+
+Bundler should respect the exact version specified for a project.
+
+If there is a lockfile (typically `Gemfile.lock`) with a `BUNDLED WITH` statement,
+it should install and use that Bundler version.
+
+Otherwise, if there is a dependency on `bundler` (in `Gemfile` or equivalent),
+it should install and use that Bundler version.
+
+This should happen transparently during the normal `bundle install`/`bundle exec` workflow.
+
+# Motivation
+
+There are many times where locking your Bundler version is useful.
+The existence of `BundlerVersionFinder` shows that, but the initial attempt
+caused innumerable problems. This is an attempt to resolve those problems.
+
+# Guide-level explanation
+
+Pin the Bundler version for a project by either:
+
+1. Commiting `Gemfile.lock`.
+2. Adding `gem "bundler", some_version_constraint` to Gemfile.
+
+If you take approach 1, you can upgrade the locked Bundler version by
+running `bundle update --bundler`.
+
+If you take approach 2, you can upgrade the locked Bundler version by
+changing the version constraint and running `bundle install`.
+
+
+## Example 1
```

Comment:
For all the example, I think it will help make them clear to also list the contents of the `Gemfile.lock` if it exists.